desktop/src/com/mygdx/game/client/ecs/system/CoordinateToPositionSystem.java

```diff
@@ -0,0 +1,36 @@
+package com.mygdx.game.client.ecs.system;
+
+import com.artemis.ComponentMapper;
+import com.artemis.annotations.All;
+import com.artemis.systems.IteratingSystem;
+import com.mygdx.game.client.ecs.component.Position;
+import com.mygdx.game.core.ecs.component.Coordinates;
+import com.mygdx.game.core.util.PositionUtil;
+import lombok.extern.java.Log;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+@Log
+@Singleton
+@All({Position.class, Coordinates.class})
+public class CoordinateToPositionSystem extends IteratingSystem {
+
+  private ComponentMapper<Position> positionMapper;
+  private ComponentMapper<Coordinates> coordinatesMapper;
+
+  @Inject
+  public CoordinateToPositionSystem() {
+  }
+
+  @Override
+  protected void process(int entityId) {
+    // todo maybe make it more based on events? does it go against the ecs philosophy?
+    var previousPosition = positionMapper.get(entityId);
+    var y = previousPosition.getPosition().y;
+    previousPosition.setPosition(
+        PositionUtil.generateWorldPositionForCoords(coordinatesMapper.get(entityId))
+    );
+    previousPosition.getPosition().y = y;
```

Comment:
czy to coś zmienia? float to typ prymitywny i nie bierzesz chyba jego referencji tylko  wartość y

Reply:
no i ogólnie nazwa previosuPosition nie pasuje bo to jest stara ale tez nowa pozycja, może entityPosition?

Reply:
zmienia to tak, że set position bierze wektora (wygenerowany_x, 0, wygenerowany_z), czyli nadpisuje się y, a y używam do tego, żeby podwyższyć trochę model wojownika, bo w moich assetach jest zakrywany przez model pola i go nie widzę

previous position może faktycznie

Reply:
Chyba nie o to mi chodziło, czy previosPosition.getPosition().y = y, rzeczywiście zmienia to y?